qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,133 @@
 import logging
-import os
 import re
-import time
-
-from autotest.client.shared import error
 
+from virttest import error_context
 from virttest import utils_misc
 
 
+@error_context.context_aware
 def run(test, params, env):
     """
     KVM windows virtio driver signed status check test:
     1) Start a windows guest with virtio driver iso/floppy
-    2) Install windows SDK in guest.
-    3) use SignTool.exe to verify whether block driver digital signed
+    2) Generate a tested driver file list.
+    3) use SignTool.exe to verify whether all drivers digital signed
 
     :param test: QEMU test object
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
-
-    def is_sdksetup_finished():
-        s, o = session.cmd_status_output("tasklist | find \"SDK\"")
-        if s:
-            return True
-        else:
-            return False
-
-    timeout = float(params.get("login_timeout", 240))
+    timeout = float(params.get("login_timeout", 360))
     session = vm.wait_for_login(timeout=timeout)
-    signtool_install = params.get("signtool_install")
-    au3_link = params.get("winsdk_au3", "autoit/winsdk.au3")
-    au3_link = os.path.join(test.bindir, au3_link)
     signtool_cmd = params.get("signtool_cmd")
     list_files_cmd = params.get("list_files_cmd")
-    drive_list = params.get("drive_list")
-    vm.copy_files_to(au3_link, "c:\\")
-    drivers = {}
-    logging.info("Install sdk in guest if signtool is not available.")
-    session.cmd_status_output(signtool_install, timeout=360)
-    # Wait until guest start install sdk.
-    time.sleep(10)
-    logging.info("Waiting for guest sdk setup ...")
-    utils_misc.wait_for(is_sdksetup_finished, timeout=1800)
-    results_all = """All the signature check log:\n"""
+    drive_volume_list = params.objects("drive_volume")
+    os_dirs = params.objects("os_dirs")
+    arch_dir = params.get("arch_dir")
     fails_log = """Failed signature check log:\n"""
+    results_all = """All the signature check log:\n"""
```

Comment:
the results_all is used to save all sign check log, it doesn't only save one line string, and the end of program, it will include many lines, so used """.